aidial_adapter_openai/utils/tokens.py

```diff
@@ -62,10 +61,11 @@ def discard_messages(
             prompt_tokens += tokenizer.calculate_tokens_per_message(message)
 
     if max_prompt_tokens < prompt_tokens:
-        raise HTTPException(
+        raise DialException(
             message=f"The token size of system messages ({prompt_tokens}) exceeds prompt token limit ({max_prompt_tokens})",
             status_code=400,
             type="invalid_request_error",
+            display_message="You exceeded amount of tokens for this chat. Please try create another one.",
```

Comment:
I would have kept the original error message and added an instruction:
`The token size of system messages ({prompt_tokens}) exceeds prompt token limit ({max_prompt_tokens}). Try reducing the length of your system messages.`

Reply:
I don't think that this message is appropriate for user.

```python

prompt_tokens = 3

for idx, message in enumerate(messages):
    if message["role"] == "system":
          kept_messages.add(idx)
          system_messages_count += 1
          prompt_tokens += tokenizer.calculate_tokens_per_message(message)
```

At this point of we calculate only tokens of system messages.

If chat has exceeded its limit, then the user cannot do anything. He can only create a new chat.

Reply:
As discussed, a user has an ability to modify system message for the existing chat conversation in DIAL Chat.